zerver/views/__init__.py

```diff
@@ -653,6 +670,30 @@ def send_registration_completion_email(email, request):
     Confirmation.objects.send_confirmation(prereg_user, email,
                                            additional_context=context)
 
+def create_realm(request):
+
+    if settings.OPEN_REALM_CREATION == False:
+        return render_to_response("zerver/realm_creation_disabled.html")
+
+    if request.method == 'POST':
+        form = create_realm_creation_form(request, user_info=request.POST)
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            send_registration_completion_email(email, request)
+            return HttpResponseRedirect(reverse('send_confirm', kwargs={'email': email}))
+        try:
+            email = request.POST['email']
+            # Note: We don't check for uniqueness
+            is_inactive(email)
+        except ValidationError:
+            return HttpResponseRedirect(reverse('django.contrib.auth.views.login') + '?email=' + urllib.parse.quote_plus(email))
```

Comment:
Can you add a comment explaining why this is the correct thing to do in the case of a ValidationError?